### PR TITLE
add more shape inference tests

### PR DESCRIPTION
**Pull request by anderspapitto**

### CODE REVIEW

File: onnx/test/test_backend_test.py [hunk 1]
```diff
@@ -37,7 +38,20 @@ def prepare(cls,
         super(DummyBackend, cls).prepare(model, device, **kwargs)
 
         # test shape inference
-        onnx.shape_inference.infer_shapes(model)
+        model = onnx.shape_inference.infer_shapes(model)
+        value_infos = {vi.name: vi for vi in itertools.chain(model.graph.value_info, model.graph.output)}
+
+        for node in model.graph.node:
+            if node.op_type in set(['RNN', 'LSTM', 'GRU', 'Reshape']):
```
Comment (bddppq) [inline onnx/test/test_backend_test.py]: let's put this blacklist/whitelist logic into a separate function

File: onnx/test/test_backend_test.py [hunk 2]
```diff
@@ -37,7 +38,20 @@ def prepare(cls,
...
+                break
+            for i, output in enumerate(node.output):
+                if node.op_type == 'Dropout' and i != 0:
```
Comment (bddppq) [inline onnx/test/test_backend_test.py]: what's special about dropout?

File: onnx/test/test_backend_test.py [hunk 3]
```diff
@@ -62,6 +75,20 @@ def supports_device(cls, device):
...
+shape_coverage_whitelist = set(
+    ['bvlc_alexnet', 'densenet121', 'inception_v1', 'inception_v2',
+     'resnet50', 'shufflenet', 'SingleRelu', 'squeezenet_old', 'vgg19', 'zfnet'])
+
+
+def do_enforce_shape_inference_coverage(model):
+    if model.graph.name not in shape_coverage_whitelist:
+        return False
+    for node in model.graph.node:
+        if node.op_type in set(['RNN', 'LSTM', 'GRU']):
```
Comment (bddppq) [inline onnx/test/test_backend_test.py]: nit: there is syntax sugar `{...}` for `set([...])`

File: onnx/test/test_backend_test.py [hunk 4]
```diff
@@ -37,7 +38,19 @@ def prepare(cls,
...
+        model = onnx.shape_inference.infer_shapes(model)
+        value_infos = {vi.name: vi for vi in itertools.chain(model.graph.value_info, model.graph.output)}
+
+        if do_enforce_shape_inference_coverage(model):
+            for node in model.graph.node:
+                for i, output in enumerate(node.output):
+                    if node.op_type == 'Dropout' and i != 0:
+                        continue
+                    assert output in value_infos
+                    tt = value_infos[output].type.tensor_type
```
Comment (linkerzhang) [inline onnx/test/test_backend_test.py]: what will happen in onnx_ml case?

Comment (anderspapitto) [reply onnx/test/test_backend_test.py]: we may have to extend this logic if we add models using the onnx_ml types to the whitelist

**Pull request merged**